lib/internal/crypto/sig.js

```diff
@@ -71,6 +72,7 @@ Sign.prototype._write = function _write(chunk, encoding, callback) {
 Sign.prototype.update = function update(data, encoding) {
   if (typeof data === 'string') {
     validateEncoding(data, encoding);
+    validateStringWellFormed(data);
```

Comment:
see other comments about encoding, this might also need adjustments and/or a check on src/ side